### PR TITLE
use disabled snapshots manager if the snapshots are disabled

### PR DESCRIPTION
**Pull request by BeniaminDrasovean**
## Reasoning behind the pull request
- If the snapshots are disabled, the node would use a static storer. When the `Snapshot()` was called, `WaitForStorageEpochChange()` was also called, and this would lead to an ERROR print because of the timeout.
  
## Proposed changes
- Use a disabled snapshots manager in case the snapshots are turned off

## Testing procedure
- Normal testing procedure
- Normal testing procedure with snapshots turned off

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

**Review by iulianpascalau — COMMENTED** [review]
Only what Darius found. GJ :+1:

File: state/disabled/disabledSnapshotsManager.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+func NewDisabledSnapshotsManager() state.SnapshotsManager {
+	return &disabledSnapshotsManger{}
+}
+
+// SnapshotState does nothing for this implementation
+func (d *disabledSnapshotsManger) SnapshotState(_ []byte, _ uint32, _ common.StorageManager) {
+}
+
+// StartSnapshotAfterRestartIfNeeded does nothing for this implementation
+func (d *disabledSnapshotsManger) StartSnapshotAfterRestartIfNeeded(_ common.StorageManager) error {
```
[the remark this answers is not in the document]
Comment (BeniaminDrasovean) [reply state/disabled/disabledSnapshotsManager.go:21]: Done

File: state/disabled/disabledSnapshotsManager.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	return nil
+}
+
+// IsSnapshotInProgress returns false for this implementation
+func (d *disabledSnapshotsManger) IsSnapshotInProgress() bool {
+	return false
+}
+
+// SetSyncer does nothing for this implementation
+func (d *disabledSnapshotsManger) SetSyncer(_ state.AccountsDBSyncer) error {
```
[the remark this answers is not in the document]
Comment (BeniaminDrasovean) [reply state/disabled/disabledSnapshotsManager.go:31]: Done.

File: state/accountsDB.go [hunk 3]
```diff
@@ -184,6 +152,9 @@ func checkArgsAccountsDB(args ArgsAccountsDB) error {
 	if check.IfNil(args.AddressConverter) {
 		return ErrNilAddressConverter
 	}
+	if check.IfNil(args.SnapshotsManager) {
+		return ErrNilSnapshotsManager
```
[the remark this answers is not in the document]
Comment (BeniaminDrasovean) [reply state/accountsDB.go:156]: Added.

**Review by iulianpascalau — DISMISSED** [review]

**Review by ssd04 — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.

**Review by ssd04 — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.

**Pull request merged**